src/java/com/rapleaf/jack/queries/GenericQueryBuilder.java

```diff
@@ -0,0 +1,176 @@
+package com.rapleaf.jack.queries;
+
+import java.io.IOException;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.SQLRecoverableException;
+import java.util.Set;
+
+import com.rapleaf.jack.BaseDatabaseConnection;
+import com.rapleaf.jack.Column;
+import com.rapleaf.jack.Table;
+import com.rapleaf.jack.queries.where_operators.IWhereOperator;
+
+public class GenericQueryBuilder {
+  private static int MAX_CONNECTION_RETRIES = 1;
+
+  private final GenericQuery genericQuery;
+  private final BaseDatabaseConnection dbConnection;
+
+  public GenericQueryBuilder(BaseDatabaseConnection dbConnection, GenericQuery genericQuery) {
```

Comment:
I remembered why I did not adopt this approach in the first place. The builder need to take two parameters: a connection, and a primary table. And I don't want to put them in the same method. So just using something like `GenericQueryBuilder.from().where(...).execute` does not solve this issue. I will try to create one derived query builder class for each database which contains the connection by default. For our use case, the builder may be `RlDbQueryBuilder`.